### DISCUSSION TITLE
PDF 辅助增强

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2721] [插件说明]
描述
---

简悦专门针对 PDF 打印的辅助增强插件。

地址
---

[https://simpread.ksria.cn/plugins/details/Lly3mkuSPz](https://simpread.ksria.cn/plugins/details/Lly3mkuSPz)

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

![image](https://user-images.githubusercontent.com/81074/173168672-a8d42f19-c427-412c-97e3-d827ae3bc9cc.png)

功能
---

1. 将导出 PDF 功能放到一级菜单
   
   ![image](https://user-images.githubusercontent.com/81074/172315360-63e0b56c-84c3-4987-98fc-05e55f6fb80e.png)
   
2. 右下角触发器可隐藏

   ![image](https://user-images.githubusercontent.com/81074/172318028-440dcc90-900d-40b1-97ca-18c23bb0c024.png)

4. 支持全局快捷键 `p p` （暂时不支持定制）

5. 当存在 [题图插件](https://simpread.ksria.cn/plugins/details/PcmPCT9rgM) 时，会提示在打印时是否隐藏/显示题图插件

    ![image](https://user-images.githubusercontent.com/81074/172315589-e90ed7dc-c91b-4062-9574-d6373d450890.png)

6. 可选 **浏览器导出方案** · **简悦同步助手基础打印方案** · **同步助手高级打印方案**

    注意：当选择使用 **简悦同步助手基础/高级打印方案** 后还是需要开启相关功能，[细节看这里](http://ksria.com/simpread/docs/#/Sync?id=pdf)。

    ![image](https://user-images.githubusercontent.com/81074/172315862-05292044-d89b-4540-b509-1ffd8f3cfb0a.png)

更新日志
---

### 1.1.6

- `Fix` 页面没有原文链接的错误（原文链接加载标题处） [#5886](https://github.com/Kenshin/simpread/issues/5886)

### 1.1.5

- `Fix` 图片尺寸设置无效的错误 [#5412](https://github.com/Kenshin/simpread/issues/5412)

### 1.1.4

-  `Fix` 当标题出现空格而导致被删除 关联 [#5656](https://github.com/Kenshin/simpread/issues/5656)

### 1.1.3

- `Fix` 当安装了 [题图](https://github.com/Kenshin/simpread/discussions/2487) 插件并且使用 [Live Editor](https://github.com/Kenshin/simpread/discussions/2852) 修改标题后，导入时仍使用旧标题的错误。

### 1.1.2

- `Fix` 当标题出现特殊字符时无法使用的错误。

### 1.1.1

- `Optimize` 当用户指定 `wkhtmltopdf` 位置时因路径忘记加 `\` 而导致的打印失败。

- `Fixed` 当图片存在标注时而导致图片尺寸出错。

### 1.1.0

- `Add` **同步助手高级打印方案**，详细说明 [请看这里](https://github.com/Kenshin/simpread/discussions/3887)。

- `Add` 右下角触发器可隐藏。

### 1.0.0

- 调用系统 PDF 时隐藏不必要的元素
- 当存在题图时是否隐藏题图？
- 可选 **浏览器导出方案** 或者 **简悦同步助手导出方案**
- 快捷键 `p p` 支持

关联
---

- [为什么说使用简悦同步助手导出的 PDF 效果最好？](https://github.com/Kenshin/simpread/discussions/2714)

- [在 Mac 端导出 PDF 出现二次编辑错位情况的解决方案（Windows 用户无此问题）
](https://github.com/Kenshin/simpread/discussions/1772)

- [如果遇到简悦 · 同步助手无法使用 PDF 等问题请看这里的解决方案](https://github.com/Kenshin/simpread/discussions/1566)

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/Lly3mkuSPz) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

**Comment by Donshii** [comment]
你好，想问一下该插件在初始化时选为了“浏览器打印方案”，如果要修改配置为“同步助手打印”，应该在哪里修改？修改配置文件的话，这里"pdf"的value应该改为什么？

```
"Lly3mkuSPz": {
    "banner": false,
    "pdf": "browser"
}
```

**Reply by Kenshin** [reply]
插件说明里面就有（就在你提问的上面...

![image](https://user-images.githubusercontent.com/81074/146662129-529b3d23-6014-4d71-a669-01be2fda9cfd.png)

**Reply by Donshii** [reply]
我一开始选了“浏览器打印方案”，现在想切换到“同步助手打印方案”，找不到这个界面的入口了。。

**Reply by Kenshin** [reply]
> 你看看你自己的是否存在 **动作栏 下面有 插件触发器** 如下图所示

![image](https://user-images.githubusercontent.com/81074/146667535-222c10d4-bc87-450c-b09e-6bddaab906cf.png)

> 如果没有的话，看这里的解决方案 https://github.com/Kenshin/simpread/discussions/2342

**Reply by Donshii** [reply]
好的，已解决

**Reply by Kenshin** [reply]
哈哈，搞定就好~

**Comment by DHS1001** [comment]
选择了PDF导出方案后关闭浏览器，再打开浏览器，导出方案又恢复成“浏览器打印方案”。
我是用的是Edge浏览器，不知是否和“关闭浏览器清除数据”的设置有关，等迟些测试一下不使用这个选项和其他浏览器。

**Reply by Kenshin** [reply]
估计是这个问题造成的，可以使用 https://github.com/Kenshin/simpread/discussions/2342 提到的 **首选方案解决**

**Reply by DHS1001** [reply]
按照**首选方案** 问题解决啦，非常感谢~

**Reply by Kenshin** [reply]
搞定就好~ 😄

**Comment by hep-cogth** [comment]


**Reply by DHS1001** [reply]
> 估计是这个问题造成的，可以使用 #2342 提到的 **首选方案解决**

可以参考楼上的解决方案试一下呀

**Comment by felix2049lou** [comment]
PDF 辅助增强选项中有没有整个网页打印（导出）成单页PDF？

**Comment by zhiWang10262418** [comment]
# 贡献个导出PDF带目录的CSS代码
将此CSS代码复制到此插件的"全局样式处"。**注意参数值后面一定带`!important`不然不生效！**
```
h1, h2, h3, h4, h5, h6 {
    display: block!important;
    margin-block-start: 0.67em;
    margin-block-end: 0.67em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    font-weight: bold;
    unicode-bidi: isolate;
}

h1 * {
  font-size: 32px!important;
}


h2  * {
  font-size: 24px!important;
}

h3  * {
  font-size: 18.72px!important; 
}

h4  * {
  font-size: 16px!important;
}

h5  * {
  font-size: 13.28px!important; 
}

h6  * {
  font-size: 11px!important; 
}
```

**Reply by Kenshin** [reply]
谢谢 👍

使用此方式如果页面有 H 标签的话，自然就会出现目录。

**Comment by zhiWang10262418** [comment]
欢迎大家共建！

发现《凤凰周刊》网页的图片名称也是H5标签。所以导出时会将图片名称也创建为标题。**希望有CSS比较专业的同学来解决**！_我只会后端，前端技术比较薄弱_。

浏览[《马伯庸：每个作家最终会踏上一条回家的路》](http://www.ifengweekly.com/detil.php?id=21681)发现的。但是这个得会员才能登陆。你就看我的图片效果吧。
<img width="1296" alt="image" src="https://github.com/user-attachments/assets/aebbf56b-0c63-4d21-af27-374c56926a81">
（原版）
<img width="1281" alt="image" src="https://github.com/user-attachments/assets/703f9ea4-2e52-4923-9fc9-cf310cd58ce3">
（简阅·阅读模式）
<img width="922" alt="image" src="https://github.com/user-attachments/assets/91f86402-bb51-4e08-a1b0-e2618d2ca88f">
（简阅·使用PDF增强插件导出后的效果）

**Reply by Kenshin** [reply]
.... 😂

我知道了，因为目录就捕捉到 H4 确实是个问题，也确实不太常见。

诸如知乎等都是仅支持到 H3 简悦又多了一级。

我考虑下优化方案。

不过这个问题跟 PDF 无关。

**Reply by zhiWang10262418** [reply]
我知道和PDF无关。
这是个别网站样式问题，属于特殊case的适配范畴。
**其实此问题对阅读体验已经只有轻度影响了**！不解决也不妨碍啥了。

**Reply by Kenshin** [reply]
没事，我先 Mark 下。

PS：这么晚还是要注意休息～

***

我建立了个 Issues 追踪这个问题 https://github.com/Kenshin/simpread/issues/6837

后续有任何讨论请在这个 Iusses 讨论 😄

**Comment by felix2049lou** [comment]
"仅将图片转换成Baseed64">"PDF辅助增强选项"＞[46-加更 02：活用文档模式，捕捉知识重点.pdf](https://github.com/user-attachments/files/18140278/46-.02.pdf)
还是没有图片
<img width="820" alt="image" src="https://github.com/user-attachments/assets/08c9962f-97a3-4759-abe6-39c64af659e2" />

https://sspai.com/post/78724

**Reply by felix2049lou** [reply]
我的simpread设置是sspai文章导出成epbu能带图片的，但是PDF高级导出是没有图片的。
再试一篇[非付费墙内的文章](https://sspai.com/post/94340),还是这样。不知道是不是我欠缺了哪一步骤。
[工作最忙的 78 天，我用 ChatGPT 考了 CFA.pdf](https://github.com/user-attachments/files/18207758/78.ChatGPT.CFA.pdf)

**Reply by Kenshin** [reply]
@felix2049lou 

PDF 确实有点问题，你先使用 Epub 方案，然后手动转换一次 PDF。

出问题的仅限 `sspai`

**Reply by felix2049lou** [reply]
[解决IOS端少数派RSS图片加载失败问题](https://renyili.org/post/%E8%A7%A3%E5%86%B3ios%E7%AB%AF%E5%B0%91%E6%95%B0%E6%B4%BErss%E5%9B%BE%E7%89%87%E5%8A%A0%E8%BD%BD%E5%A4%B1%E8%B4%A5%E9%97%AE%E9%A2%98/)
里面提到的解决原理能在simpread
里面实现吗？
